src/dynarmic/backend/x64/emit_x64_vector.cpp

```diff
@@ -1803,6 +1803,45 @@ void EmitX64::EmitVectorLogicalShiftRight64(EmitContext& ctx, IR::Inst* inst) {
 }
 
 void EmitX64::EmitVectorLogicalVShift8(EmitContext& ctx, IR::Inst* inst) {
+    if (code.HasHostFeature(HostFeature::AVX512_Ortho | HostFeature::AVX512BW | HostFeature::GFNI)) {
+        auto args = ctx.reg_alloc.GetArgumentInfo(inst);
+        const Xbyak::Xmm result = ctx.reg_alloc.UseScratchXmm(args[0]);
+        const Xbyak::Xmm left_shift = ctx.reg_alloc.UseScratchXmm(args[1]);
+        const Xbyak::Xmm tmp = ctx.reg_alloc.ScratchXmm();
+
+        const Xbyak::Opmask negative_mask = k1;
+        code.pxor(tmp, tmp);
+        code.vpcmpb(negative_mask, left_shift, tmp, CmpInt::LessThan);
+
+        // Reverse bits of negative-shifts
+        code.vgf2p8affineqb(result | negative_mask, result, code.BConst<64>(xword_b, 0x8040201008040201), 0);
```

Comment:
`xmm0` is guaranteed to be a free scratch register.

```suggestion
        code.vmovaps(xmm0, code.BConst<64>(xword_b, 0x8040201008040201));
        code.vgf2p8affineqb(result | negative_mask, result, xmm0, 0);
```